cli/cli/commands/port/print/print.go

```diff
@@ -113,13 +131,32 @@ func run(
 		)
 	}
 
-	fullUrl := fmt.Sprintf("%v:%v", ipAddress, publicPort.GetNumber())
-	maybeApplicationProtocol := publicPort.GetMaybeApplicationProtocol()
-
-	if maybeApplicationProtocol != "" {
-		fullUrl = fmt.Sprintf("%v://%v", maybeApplicationProtocol, fullUrl)
+	fullUrl, err := formatOutput(format, ipAddress, publicPort)
+	if err != nil {
+		return stacktrace.Propagate(err, "Couldn't format the output according to formatting string '%v'", format)
 	}
-
 	out.PrintOutLn(fullUrl)
 	return nil
 }
+
+func formatOutput(format string, ipAddress string, spec *services.PortSpec) (string, error) {
+	parts := strings.Split(format, ",")
+	var resultParts []string
+	for _, part := range parts {
+		switch part {
+		case "protocol":
```

Comment:
Same - can we make these named constants so we're DRY?

Reply:
Will do